lib/appsignal/hooks/sidekiq.rb

```diff
@@ -12,6 +12,9 @@ def dependencies_present?
       end
 
       def install
+        require "sidekiq/api" if Appsignal.config[:enable_minutely_probes]
```

Comment:
If the `enable_minutely_probes` is disables, we don't require sidekiq, but we do add the probe. Doesn't this break in that case? I.e. shouldn't we only add the probe if the minutely probe is enables?

Reply:
The minutely probe doesn't get called if `enable_minutely_probes` is `false`.
I can add it to the same if-statement if that feels more safe. Or even add it to the Probe initializer?

Reply:
I would prefer adding it to the same if statement. So we never add the probe if it is disabled.

Reply:
I don't know.. Maybe it's better to keep that responsibility with the `Minutely` class? I wouldn't want to check the config for every hook in which we add a minutely probe. Easily forgotten and adds complexity to every hook.

We could change it to support uninitialized classes, have the Minutely class initialize the probe when they `enable_minutely_probes` is true. Then we can move the require to the Probe's initializer and not have if-statement for every probe added to the probes list.

A solution to support initialized and uninitialized probes would be something like this:

```diff
diff --git a/lib/appsignal/hooks/sidekiq.rb b/lib/appsignal/hooks/sidekiq.rb
index 670a56f0..c8557d01 100644
--- a/lib/appsignal/hooks/sidekiq.rb
+++ b/lib/appsignal/hooks/sidekiq.rb
@@ -12,7 +12,6 @@ module Appsignal
       end

       def install
-        require "sidekiq/api" if Appsignal.config[:enable_minutely_probes]
         Appsignal::Minutely.probes << SidekiqProbe.new

         ::Sidekiq.configure_server do |config|
@@ -25,6 +24,7 @@ module Appsignal

     class SidekiqProbe
       def initialize
+        require "sidekiq/api"
         @cache = {}
       end

diff --git a/lib/appsignal/minutely.rb b/lib/appsignal/minutely.rb
index 15a5670f..389efbff 100644
--- a/lib/appsignal/minutely.rb
+++ b/lib/appsignal/minutely.rb
@@ -10,12 +10,20 @@ module Appsignal
         @@probes ||= []
       end

+      def probe_instances
+        @@probe_instances ||= []
+      end
+
       def start
+        probes.each do |probe_class|
+          instance = probe_class.respond_to?(:new) ? probe_class.new : probe_class
+          probes_instances << instance
+        end
         Thread.new do
           loop do
             logger = Appsignal.logger
             logger.debug("Gathering minutely metrics with #{probes.count} probes")
-            probes.each do |probe|
+            probes_instances.each do |probe|
               begin
                 name = probe.class.name
                 logger.debug("Gathering minutely metrics with #{name} probe")
```

Anyone else have an opinion about this @matsimitsu @thijsc @jeffkreeftmeijer ?

Reply:
Ideally customers can create probes, so adding probes while not enabling them should be possible and not break the integration

Reply:
Might be nice to add something to the probe class. Separate `install_probes` method that only gets called if enabled?

Reply:
We've fixed that in https://github.com/appsignal/appsignal-ruby/pull/474. That allows probes to be initialized when the Minutely thread is started and not beforehand. What I've described in the PR description is how we'll implement it in this PR once that's merge.